src/thunderbots/software/ai/hl/stp/tactic/shoot_goal_tactic.cpp

```diff
@@ -0,0 +1,161 @@
+#include "ai/hl/stp/tactic/shoot_goal_tactic.h"
+
+#include "ai/hl/stp/action/move_action.h"
+#include "ai/hl/stp/evaluation/calc_best_shot.h"
+#include "ai/hl/stp/evaluation/intercept.h"
+
+ShootGoalTactic::ShootGoalTactic(const Field &field, const Team &friendly_team,
+                                 const Team &enemy_team, const Ball &ball,
+                                 double min_percent_net_open,
+                                 std::optional<Point> chip_target, bool loop_forever)
+    : field(field),
+      friendly_team(friendly_team),
+      enemy_team(enemy_team),
+      ball(ball),
+      min_percent_net_open(min_percent_net_open),
+      chip_target(chip_target),
+      has_shot_available(false),
+      Tactic(loop_forever)
+{
+}
+
+std::string ShootGoalTactic::getName() const
+{
+    return "Shoot Goal Tactic";
+}
+
+void ShootGoalTactic::updateParams(const Field &field, const Team &friendly_team,
+                                   const Team &enemy_team, const Ball &ball)
+{
+    this->field         = field;
+    this->friendly_team = friendly_team;
+    this->enemy_team    = enemy_team;
+    this->ball          = ball;
+}
+
+double ShootGoalTactic::calculateRobotCost(const Robot &robot, const World &world)
+{
+    auto ball_intercept_opt =
+        Evaluation::findBestInterceptForBall(world.ball(), world.field(), robot);
+    double cost = 0;
+    if (ball_intercept_opt)
+    {
+        // If we can intercept the ball, use the distance to the intercept point.
+        // We normalize with the total field length so that robots that are within the
+        // field have a cost less than 1
+        cost = (ball_intercept_opt->first - robot.position()).len() /
+               world.field().totalLength();
+    }
+    else
+    {
+        // If we can't intercept the ball, just use the distance to the ball's current
+        // position. We normalize with the total field length so that robots that are
+        // within the field have a cost less than 1
+        cost = (world.ball().position() - robot.position()).len() /
+               world.field().totalLength();
+    }
+
+    return std::clamp<double>(cost, 0, 1);
+}
+
+bool ShootGoalTactic::hasShotAvailable() const
+{
+    return has_shot_available;
+}
+
+bool ShootGoalTactic::isEnemyAboutToStealBall() const
+{
+    // TODO: replace with get all robots except goalie?
```

Comment:
TODO?

Reply:
Can't be resolved until the changes from my defense PR are merged. I'll resolve it after.